pkg/kubectl/describe.go

```diff
@@ -528,7 +529,30 @@ func describeContainers(pod *api.Pod, out io.Writer) {
 
 		fmt.Fprintf(out, "    Ready:\t%v\n", printBool(status.Ready))
 		fmt.Fprintf(out, "    Restart Count:\t%d\n", status.RestartCount)
+		fmt.Fprintf(out, "    Environment Variables:\t\n")
+		for _, e := range container.Env {
+			if e.ValueFrom != nil && e.ValueFrom.FieldRef != nil {
+				valueFrom := envValueFrom(pod, e)
+				fmt.Fprintf(out, "        \t%s\t = %s (%s:%s)\n", e.Name, valueFrom, e.ValueFrom.FieldRef.APIVersion, e.ValueFrom.FieldRef.FieldPath)
+			} else {
+				fmt.Fprintf(out, "        \t%s\t = %s\n", e.Name, e.Value)
+			}
+		}
+	}
+}
+
+func envValueFrom(pod *api.Pod, e api.EnvVar) string {
+	internalFieldPath, _, err := api.Scheme.ConvertFieldLabel(e.ValueFrom.FieldRef.APIVersion, "Pod", e.ValueFrom.FieldRef.FieldPath, "")
+	if err != nil {
+		return err.Error()
 	}
+
+	valueFrom, err2 := fieldpath.ExtractFieldPathAsString(pod, internalFieldPath)
```

Comment:
It doesn't look like custom fieldpaths are supported by `ExtractFieldPathAsString` 
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/fieldpath/fieldpath.go#L29

Did you verify that the test you added has expected output, not an error string that happens to contain expected values? This goes back to above question re, reasonableness of using errorstring in place of envvar value.

Also, a nit: no need to create a new `err2` var, you can reuse `err`.